xdsl/dialects/riscv.py

```diff
@@ -127,15 +138,18 @@ class Register:
         "ft10": 30,
         "ft11": 31,
     }
+    ABI_INDEX_BY_NAME = RV32F_INDEX_BY_NAME
 
 
-@irdl_attr_definition
-class RegisterType(Data[Register], TypeAttribute):
+RegisterT = TypeVar("RegisterT", bound="Register")
+
+
+class RegisterType(GenericData[RegisterT], TypeAttribute):
```

Comment:
@math-fehr now that I think about it, maybe this should be `RegisterAttr` instead?